### PR TITLE
Update PHP Version From 8.0 to 8.2

### PR DESCRIPTION
**Pull request by vahidbazzaz**
## WIP

This PR continues the effort to add PHP 8.2 support following the upgrade to 8.0 in [PR #577](https://github.com/boxwise/dropapp/pull/577). 

Changes included in this PR:

- Updated the development Dockerfile to PHP 8.2 and Xdebug to 3.3.1.
- Updated CircleCI PHP images to 8.2.
- Silenced Smarty warnings.
- Updated the App Engine runtime to PHP 8.2.
- Set Rector PHP version to 8.2.
- Refactored using Rector.
- Replaced `strftime`.
- Fix Smarty templates (add custom php function modifiers, variable naming changes and etc.)

### CODE REVIEW

**Comment by vahidbazzaz** [comment]
[update php version from 7.4 for dropapp](https://trello.com/c/FgTSr1Mz/1364-update-php-version-from-74-for-dropapp)

File: library/lib/smarty.php [hunk 1]
```diff
@@ -41,4 +44,51 @@ public function fetch($template = null, $cache_id = null, $compile_id = null, $p
...
     }
+
+    // Register custom modifiers
+    private function registerCustomModifiers()
+    {
+        $this->registerPlugin(Smarty::PLUGIN_MODIFIER, 'is_array', 'is_array_modifier');
+        $this->registerPlugin(Smarty::PLUGIN_MODIFIER, 'array_slice', 'array_slice_modifier');
+        $this->registerPlugin(Smarty::PLUGIN_MODIFIER, 'implode', 'implode_modifier');
+        $this->registerPlugin(Smarty::PLUGIN_MODIFIER, 'implode_array', 'implode_array_modifier');
+        $this->registerPlugin(Smarty::PLUGIN_MODIFIER, 'intval', 'intval_modifier');
```
Comment (HaGuesto) [inline library/lib/smarty.php:55]: we need an explode modifier for stock_overview.

see templates/cms_list.tpl in line 145

**Review by HaGuesto — COMMENTED** [review]
Next step: query all php functions and modifiers in the smarty templates:
- query all using the piping operator and
- query a regexp with { ( ) }

**Review by HaGuesto — APPROVED** [review]
## Current problem

When checking the query for stock overview we get different results if we run the query in our docker environment or in the staging environment. We do not know yet why.
In staging we get currently the correct results, locally many rows are duplicate.

We are merging this PR to check if this behavior persists.

## Other remarks 

- db_array and sb_simplearray returning empty arrays now instead of false before: maybe this will create issues when validating them.
- in start.php I'm not sure if the DateTime formatting works correctly. Page needs to be checked on production data.

**Pull request merged**